src/irr/video/COpenGLFunctionTable.h

```diff
@@ -40,3733 +15,3084 @@ wywali
+			,glXSwapIntervalEXT
+#endif
+#if defined(GLX_MESA_swap_control) && !defined(_IRR_COMPILE_WITH_SDL_DEVICE_)
+			,glXSwapIntervalMESA
+#endif
+		);
+		IRR_SYSTEM_DECLARE_DYNAMIC_FUNCTION_CALLER_CLASS(GLprogram, OpenGLFunctionLoader
+			,glEnablei
+			,glDisablei
+			,glProgramParameteri
+			,glPatchParameterfv
+			,glPatchParameteri
+		);
 
 
-		inline void COpenGLFunctionTable::extGlBindImageTexture(GLuint index, GLuint texture, GLint level, GLboolean layered, GLint layer, GLenum access, GLenum format)
-		{
-			if (pGlBindImageTexture)
-				pGlBindImageTexture(index, texture, level, layered, layer, access, format);
-		}
 
 
-		inline void COpenGLFunctionTable::extGlBindImageTextures(GLuint first, GLsizei count, const GLuint* textures, const GLenum* formats)
-		{
-			if (pGlBindImageTextures)
-				pGlBindImageTextures(first, count, textures);
-			else
-			{
-				for (GLsizei i = 0; i < count; i++)
-				{
-					if (!textures || textures[i] == 0u)
-						extGlBindImageTexture(first + i, 0u, 0u, GL_FALSE, 0, GL_READ_WRITE, GL_R8);
-					else
-						extGlBindImageTexture(first + i, textures[i], 0, GL_TRUE, 0, GL_READ_WRITE, formats[i]);
-				}
-			}
-		}
+		//GLCapabilities
 
-		inline GLuint64 COpenGLFunctionTable::extGlGetTextureHandle(GLuint texture)
-		{
-			if (pGlGetTextureHandleARB)
-				return pGlGetTextureHandleARB(texture);
-			else if (pGlGetTextureHandleNV)
-				return pGlGetTextureHandleNV(texture);
-			return 0ull;
-		}
 
-		inline GLuint64 COpenGLFunctionTable::extGlGetTextureSamplerHandle(GLuint texture, GLuint sampler)
-		{
-			if (pGlGetTextureSamplerHandleARB)
-				return pGlGetTextureSamplerHandleARB(texture, sampler);
-			else if (pGlGetTextureSamplerHandleNV)
-				return pGlGetTextureSamplerHandleNV(texture, sampler);
-			return 0ull;
-		}
 
-		inline void COpenGLFunctionTable::extGlMakeTextureHandleResident(GLuint64 handle)
-		{
-			if (pGlMakeTextureHandleResidentARB)
-				return pGlMakeTextureHandleResidentARB(handle);
-			else if (pGlMakeTextureHandleResidentNV)
-				return pGlMakeTextureHandleResidentNV(handle);
-		}
 
-		inline void COpenGLFunctionTable::extGlMakeTextureHandleNonResident(GLuint64 handle)
-		{
-			if (pGlMakeTextureHandleNonResidentARB)
-				return pGlMakeTextureHandleNonResidentARB(handle);
-			else if (pGlMakeTextureHandleNonResidentNV)
-				return pGlMakeTextureHandleNonResidentNV(handle);
-		}
 
-		inline GLuint64 COpenGLFunctionTable::extGlGetImageHandle(GLuint texture, GLint level, GLboolean layered, GLint layer, GLenum format)
+	protected:
+		// constructor
+		COpenGLFunctionTable() 
 		{
-			if (pGlGetImageHandleARB)
-				return pGlGetImageHandleARB(texture, level, layered, layer, format);
-			else if (pGlGetImageHandleNV)
-				return pGlGetImageHandleNV(texture, level, layered, layer, format);
-			return 0ull;
+			if(SDL_GL_LoadLibrary(nullptr);) //nullptr to load the default OpenGL library
+				//log error
+			if(SDL_VideoInit())
```

Comment:
its not the function table that should do it, it should be the library loader instead